### PR TITLE
feat: auto-scale

### PR DESCRIPTION
**Pull request by ChaikaS**

### CODE REVIEW

File: components/scg/kb/sources/controls/ui_scg_control_tool_autosize.scs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   		<= nrel_sc_text_translation:	
+	     ..t1
+	    	(*
+        	-> [Используйте эту функцию для автоматического подбора масштаба] (* <- lang_ru;; *);;
+        	*);
+		..t2
+		(*
+		-> [Use this function to auto-size a picture] (* <- lang_en;; *);;
+		*);;
+    *);;
```
Comment (smousy) [inline components/scg/kb/sources/controls/ui_scg_control_tool_autosize.scs:22]: add newline

File: components/scg/src/scg-render.js [hunk 2]
```diff
@@ -347,7 +348,7 @@ SCg.Render.prototype = {
 
     // -------------- update objects --------------------------
     updateObjects: function () {
-
+
```
Comment (smousy) [inline components/scg/src/scg-render.js:351]: remove

File: components/scg/src/scg.js [hunk 3]
```diff
@@ -785,6 +789,11 @@ SCg.Editor.prototype = {
             await updateConfirmedData();
         });
 
+        this.toolAutoSize().click(function () {
+            const [conteinerWidth, conteinerHeight] = self.scene.getContainerSize();
+            self.scene.render._changeContainerTransform([conteinerWidth * 0.3, conteinerHeight * 0.3], 0.5)
```
Comment (smousy) [inline components/scg/src/scg.js:794]: why 0.3 and 0.5, please use constants

File: client/js/Core/componentsandbox.js [hunk 4]
```diff
@@ -278,6 +282,23 @@ SCWeb.core.ComponentSandbox.prototype.updateContent = async function (contentTyp
         for (let triple of result) {
             self.eventStructUpdate(true, triple.get("src").value, triple.get("edge").value);
         }
+
+        setTimeout(() => {
+            if (self.scene) {
+                const [conteinerWidth, conteinerHeight] = self.scene.getContainerSize();
+                const scgHeight = self.scene.render.d3_container[0][0].getBoundingClientRect().height + (conteinerHeight/2 + 350);
```
Comment (smousy) [inline client/js/Core/componentsandbox.js:289]: please use constants instead of numbers

**Review by smousy — CHANGES_REQUESTED** [review]

**Review by AlexandrZagorskiy — APPROVED** [review]
Works well

File: client/static/components/css/bootstrap-override.css [hunk 5]
```diff
@@ -0,0 +1,23 @@
+.btn-group-vertical>.btn, .btn-group-vertical>.btn-group, .btn-group-vertical>.btn-group>.btn {
```
Comment (MikhailSadovsky) [inline client/static/components/css/bootstrap-override.css:1]: This file shouldn't be committed and should be added to gitignore

File: client/js/Core/componentsandbox.js [hunk 6]
```diff
@@ -278,6 +282,27 @@ SCWeb.core.ComponentSandbox.prototype.updateContent = async function (contentTyp
...
+                const graphAspectRatio = scgHeight / scgWidth;
+                
+                const scale = containerAspectRatio < graphAspectRatio ? 
+                conteinerWidth / scgWidth : 
+                    conteinerHeight / scgHeight;
+                
+                
+                self.scene.render._changeContainerTransform([conteinerWidth * deltaTranslate, conteinerHeight * deltaTranslate], scale);
+            }
+        },300)
```
Comment (MikhailSadovsky) [inline client/js/Core/componentsandbox.js:305]: Please move to constant

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

File: components/scg/kb/sources/controls/ui_scg_control_tool_autosize.scs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (NikitaZotov) [inline components/scg/kb/sources/controls/ui_scg_control_tool_autosize.scs:23]: Add newline

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**